src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1491,6 +1503,21 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) ([]*grpc.Storag
 	return epheStorages, nil
 }
 
+func parseEmptyDirSize(mountPoint string) (string, error) {
+	_, _, options, err := utils.GetMountInfo(mountPoint)
+	if err != nil {
+		return "", err
+	}
+
+	for i := range options {
+		if strings.HasPrefix(options[i], "size=") {
+			return options[i], nil
+		}
+	}
+	// don't return error even size option not found.
```

Comment:
Is it a possible outcome?

Reply:
I am not sure, but I'm curious.  If you make a tmpfs mount w/ out specifying size, the default is half of the machine's RAM.  Is that something that is reflected from mount info? Or is no size option specified?

Reply:
Yes, the default size is 1/2 of the host.

In case of runc, the mount info for

```yaml
  volumes:
  - name: memory-empty-vol
    emptyDir:
      medium: Memory
  - name: memory-empty-vol-sized
    emptyDir:
      medium: Memory
      sizeLimit: "50M"
```

Will be:

```
tmpfs on /var/lib/kubelet/pods/b49f8a51-1dbb-4f88-87af-6224f2d54b14/volumes/kubernetes.io~empty-dir/memory-empty-vol type tmpfs (rw,relatime,size=4776732k)
tmpfs on /var/lib/kubelet/pods/b49f8a51-1dbb-4f88-87af-6224f2d54b14/volumes/kubernetes.io~empty-dir/memory-empty-vol-sized type tmpfs (rw,relatime,size=48832k)
tmpfs on /var/lib/kubelet/pods/b49f8a51-1dbb-4f88-87af-6224f2d54b14/volumes/kubernetes.io~projected/kube-api-access-mrljd type tmpfs (rw,relatime,size=4776732k)
```

Where 4776732k is half of the host.

Reply:
@liubin - sorry our convo is taking a while; latency in pings!

I understand what runc does and the default behavior -- my question is - do you expect that when sizeLimit is not set in the pod spec that you'd find size option specified when parsing the empty dir size? On our side we didn't see it set, and I want to double check!